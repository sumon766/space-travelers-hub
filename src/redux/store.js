import { configureStore } from '@reduxjs/toolkit';
import rocketReducer from './rockets/rocketSlice';
import missionReducer from './missions/missionSlice';
import dragonsReducer from './dragons/dragonsSlice';

const store = configureStore({
  reducer: {
    Rocket: rocketReducer,
    Mission: missionReducer,
    dragons: dragonsReducer,
  },
});

export default store;
