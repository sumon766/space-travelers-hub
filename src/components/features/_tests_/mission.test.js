import React from 'react';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import store from '../../../redux/store';
import '@testing-library/jest-dom';
import Missions from '../../rockets/missions';

describe('performing test for missions', () => {
  it('testing if mission page really matches', () => {
    const missionRender = renderer
      .create(
        <Provider store={store}>
          <Missions />
        </Provider>,
      )
      .toJSON();
    expect(missionRender).toMatchSnapshot();
  });
});
