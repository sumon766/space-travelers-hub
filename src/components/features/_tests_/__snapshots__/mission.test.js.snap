// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`performing test for missions testing if mission page really matches 1`] = `
<div
  className="missions-area"
>
  <table
    className="missions-table"
  >
    <thead>
      <tr>
        <th>
          Mission
        </th>
        <th>
          Description
        </th>
        <th>
          Status
        </th>
        <th>
          Join/Leave Mission
        </th>
      </tr>
    </thead>
    <tbody />
  </table>
</div>
`;
